### PR TITLE
Ensure merge order

### LINKED ISSUE #30: Ensure merge order of PRs in same hourly batch
The current merging strategy does not guarantee merge order for PRs in the same hourly batch. However, reviewers may intend for one outstanding PR to be merged before another. The bot should ensure that there is some clear ordering to how PRs in the same batch are merged such that reviewers can plan accordingly. This could be based on the PR number (lowest first) or the chronological order in which PRs were approved.

### PR DESCRIPTION
**Pull request by mbestavros**
This PR rewrites the `merge` action to merge PRs in ascending order of the time of last approval.

It's a little messy, I haven't tested it yet, and it may need changes, so I'm submitting in draft form.

Resolves #30.

### CODE REVIEW

**Comment by lkatalin** [comment]
This looks like it would work. Is there a way to test it before using it? (Probably not?)

I'm wondering if there is any way to build up the `approved` list in real-time instead of going through, getting the info, and sorting. That would be more efficient, but seems harder, and I'm not at all familiar with what our current automation can or can't do.

**Comment by mbestavros** [comment]
@npmccallum My concern with the format you suggested is that `pr.mergeable` and `pr.mergeable_state` might change after merging another PR, so it should really be checked right before merge (instead of pre-checked and then merged based on that).

Regardless, I've still made changes, and all printing now happens in one loop. Instead of only collecting timestamps for those known to be mergeable, the script now collects for _all_ PRs and then sorts them. Then, the script checks for mergeability and merges, and prints status.

File: enarx-merge [hunk 1]
```diff
@@ -6,18 +6,39 @@ import enarxbot
...
     pr = issue.as_pull_request()
 
-    # If all merge criteria pass, merge the PR.
+    # Get the latest PR review
+    latest_review = ""
+    for i in pr.get_reviews():
+        if i.submitted_at.isoformat() > latest_review:
+            latest_review = i.submitted_at.isoformat()
+
+    prs_timestamped[latest_review] = (pr, pr_identifier)
```
Comment (npmccallum) [inline enarx-merge:24]:
```suggestion
    prs_timestamped[latest_review] = (issue, pr)
```
Then you can construct the `pr_identifier` later while also having the full issue object for later script modifications.

File: enarx-merge [hunk 2]
```diff
@@ -6,18 +6,39 @@ import enarxbot
 # Initialize Github
 github = enarxbot.connect()
 
-# Print column keys.
-print(" Merged? | Repository + PR number | Mergeable? | State  ")
-
+# Timestamp all PRs.
+prs_timestamped = {}
 for issue in github.search_issues("org:enarx state:open is:public is:pr review:approved"):
```
Comment (npmccallum) [inline enarx-merge:11]:
```suggestion
for issue in github.search_issues("org:enarx state:open is:public is:pr review:approved sort:updated-asc"):
```
Could we just do this and avoid all the other changes?

**Pull request merged**